### PR TITLE
Add formatCurrencyAmount util

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Format the input amount with the currency.
 
For example, if using `V2V3_CURRENCY_USD` with an amount of 550.1 the format
will be `"$550.1 USD"`.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2068** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2068" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2068?utm_source=stack-comment).

File: src/utils/formatCurrencyAmount.ts
Comment (tomquirk) [inline src/utils/formatCurrencyAmount.ts:18]: **suggestion** WDYT about using standard currency notation here instead: https://www.investopedia.com/terms/c/currency-symbol.asp#:~:text=If%20the%20currency%20type%20is,such%20as%20the%20Aruban%20Florin.

```suggestion
  const currencyPrefix = currency === V2V3_CURRENCY_USD ? 'US$' : ''
  const currencySuffix = currency === V2V3_CURRENCY_USD ? '' : 'ETH'
```

Or even:

```suggestion
  const currencyPrefix = currency === V2V3_CURRENCY_USD ? 'US$' : 'Ξ'
```

Comment (wraeth-eth) [reply src/utils/formatCurrencyAmount.ts:26]: I'm happy either way, just unsure if Ξ might trip people up if theyre not familiar with it

File: src/utils/formatCurrencyAmount.ts
Comment (benschac) [inline src/utils/formatCurrencyAmount.ts:5]: Could this arg be `number | string` or something else https://lodash.com/docs/4.17.15#isInteger.

File: src/utils/formatCurrencyAmount.ts
Comment (benschac) [inline src/utils/formatCurrencyAmount.ts:6]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger Could we nix  lodash?

File: src/utils/formatCurrencyAmount.ts
Comment (wraeth-eth) [reply src/utils/formatCurrencyAmount.ts:5]: potentially but lets wait till we have a usecase I reckon

**Review by tomquirk — APPROVED** [review]
@wraeth-eth will pre-approve this, pending the `US$` changes discussed in discord 🚀

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2068?utm_source=stack-comment).


Job Reference: TLpS7mRK611WyBbtlMOK

**Pull request merged**
